docs/src/architecture/08_concepts/cardano_chain_follower/follower_instances.md

```diff
@@ -0,0 +1,122 @@
+# Cardano Chain Follower Instances
+
+This document details how the individual chain follower instances are organized within the Hermes runtime extension.
+
+## Basic Principles
+
+There are 3 primary cardano networks, Mainnet, Preprod and Preview.
+We will often be syncing from TIP or Genesis, and after synchronization we will be syncing from Tip eventually.
+
+A node follower is a reasonably heavy process, and every connection to a relay node also adds load to the relay node itself.
+
+It is not efficient or necessary to have multiple followers syncing from the same point as the data stream is identical.
+
+## Pool of Node followers
+
+We will maintain a Pool of active node followers.
+When a wasm module in an application subscribes to a Node followers data stream, IF there is a pre-existing node follower
+which can serve its needs, then it should simply receive the events from that stream.
+
+It will not create a new stream for every subscription.
+
+When the last subscriber to a Node followers connection/stream unsubscribes, it will stop the connection.
+
+This means that there is not one follower per application, there is one follower per stream type.
+
+For example,  if a wasm module subscribes to a stream of events from Genesis, and Tip, it will have two active streams.
```

Comment:
A tip follower is a special case in the sense that one instance can be shared by all subscriber modules that want to subscribe to tip events.

Reply:
Yes,  but its also just a special case of multiple subscriber modules wanting to subscribe to the same point in the stream.
It's just that this is going to be highly likely so we should accommodate it now.
But the general case is anytime two followers end up on the same block on the same network, they can be collapsed into a single follower.